cmd/clusterctl/client/cluster/topology.go

```diff
@@ -251,15 +267,21 @@ func (t *topologyClient) prepareInput(ctx context.Context, in *DryRunInput, apiR
 
 // setMissingNamespaces sets the object to the current namespace on objects
 // that are missing the namespace field.
-func (t *topologyClient) setMissingNamespaces(in *DryRunInput) error {
-	currentNamespace := metav1.NamespaceDefault
-	// If a cluster is available use the current namespace as defined in its kubeconfig.
-	if err := t.proxy.CheckClusterAvailable(); err == nil {
-		currentNamespace, err = t.proxy.CurrentNamespace()
-		if err != nil {
-			return errors.Wrap(err, "failed to get current namespace")
+func (t *topologyClient) setMissingNamespaces(in *TopologyPlanInput) error {
```

Comment:
```suggestion
func (t *topologyClient) setMissingNamespaces(string namespace) error {
```
Can we just use the namespace string as parameter? Just makes it way clearer what this method works with (also makes l.272 redundant I think)

Reply:
I need `in *TopologyPlanInput` here. Because this functions loops over `in.Objs` to set the namespaces on them. The alternative would be `setMissingNamespaces(string namespace, objs []*unstructured.Unstructured)` and this seems resonable.

Reply:
Oh sorry my bad, missed that :/ Feel free to resolve